src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -692,6 +692,22 @@ public function selectLink($value)
     }
 
     /**
+     * Selects images by alt value.
+     *
+     * @param string $value The image alt
+     *
+     * @return Crawler A new instance of Crawler with the filtered list of nodes
+     *
+     * @api
+     */
+    public function selectImage($value)
+    {
+        $xpath = sprintf('descendant-or-self::img[contains(concat(\' \', normalize-space(string(@alt)), \' \'), %s)]', static::xpathLiteral(' '.$value.' '));
```

Comment:
you don't need this concatenation stuff at all for the alt attribute. Adding a space at the beginning and the end of the class attribute and around the value is the simple way to search elements with a given class in the space separated list. the alt attribute is not a space separated list in HTML